runtime/altair/src/lib.rs

```diff
@@ -1573,6 +1574,15 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl runtime_common::apis::LoansApi<Block, PoolId, Balance> for Runtime {
+		fn pool_valuation(pool_id: PoolId) -> Option<Balance>{
+			match pallet_loans::Pallet::<Runtime>::update_nav_of_pool(pool_id) {
+				Ok((_, value)) => Some(value),
+				Err(_) => None,
+			}
+		}
+	}
+
```

Comment:
The api should do something like: 

```rust
	impl runtime_common::apis::LoansApi<Block, PoolId, InstanceId, Balance> for Runtime {
		fn nav(id: PoolId) -> Option<Balance> {
			// TODO: Error out if pool does not exists in the loans-pallet
			if !pallet_pools::Pool::<Runtime>::get(id).is_some() {
				return None;
			}
			pallet_loans::Pallet::<Runtime>::update_nav_of_pool(id)
				.ok()
				.map(|(latest, _)| latest.into())
		}

		fn max_borrow_amount(id: PoolId, loan_id: InstanceId) -> Option<Balance> {
			let now = <pallet_timestamp::Pallet::<Runtime> as UnixTime>::now().as_secs();
			pallet_loans::Loan::<Runtime>::get(id, loan_id)
				.map(|loan_details| loan_details.max_borrow_amount(now).into())
		}
	}
```

Reply:
Naming of course not right here. I just took the code from the old branch I implemented this on.

Reply:
Best would be to do all the checks that this section does https://github.com/centrifuge/centrifuge-chain/blob/4a10ae7a7eaff0062e7aca3bcba4b4129e909bc8/pallets/loans/src/functions.rs#L357-L393 also in the API code.